lib/reline/config.rb

```diff
@@ -133,26 +135,26 @@ def read(file = nil)
 
   def key_bindings
     # The key bindings for each editing mode will be overwritten by the user-defined ones.
-    kb = @key_actors[@editing_mode_label].default_key_bindings.dup
-    kb.merge!(@additional_key_bindings[@editing_mode_label])
-    kb.merge!(@oneshot_key_bindings)
-    kb
+    Reline::KeyActor::Composite.new([@oneshot_key_bindings, @additional_key_bindings[@editing_mode_label], @default_key_bindings[@editing_mode_label]])
   end
 
   def add_oneshot_key_binding(keystroke, target)
-    @oneshot_key_bindings[keystroke] = target
+    # IRB sets invalid keystroke [Reline::Key]. We should ignore it.
```

Comment:
Have we removed that from IRB?

Reply:
Not yet.
Even if we remove it, we need to support `[Reline::Key.new(nil, 0xE4, true)]` for a while, not to break an environment that new Reline is installed but IRB is not updated.

Reply:
Yes I'm aware of that and I'm not arguing we need to change the approach here 🙂  I just hadn't see any related PR on IRB so want to confirm.

Reply:
I opened https://github.com/ruby/irb/pull/963